header-rewrite-filter/header_processor.cc

```diff
@@ -172,7 +172,7 @@ namespace HeaderRewriteFilter {
                 case Utility::MatchType::Exact:
                     string_to_compare = std::string(*(start + 4));
                     source_ = *(start + 1);
-                    matcher_ = [string_to_compare](absl::string_view source) { return (source.compare(string_to_compare) == 0); };
+                    matcher_ = [string_to_compare](std::string source) { return (source.compare(string_to_compare) == 0); };
```

Comment:
Was there an issue with the `string_view` here? This function is very short and returns almost immediately, and doesn't keep `source` around to use later. It doesn't seem possible for it to outlive the original string

Reply:
I did run into an issue with keeping it a `string_view`. The match comparison would return unexpected and nondeterministic values when I used `string_view`, and the issue went away when I switched to `string` -- maybe it could be because I'm storing the lambda function to be used later on when the scope has changed?

Reply:
The parameter isn't evaluated when you store the lambda. `string_to_compare` is stored in the lambda, but `source` isn't. That's a bit worrisome, that makes me think the memory corruption you saw was caused by something else. Unless I'm missing something. 

We can figure it out if it shows up again, I guess